sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -805,6 +805,22 @@ abstract class CSVSuite
     }
   }
 
+  test("SPARK-37575: null values should not reflect to any characters by default") {
+    val litNull: String = null
+    val data = Seq(("Tesla", litNull, ""))
+    withTempPath { path =>
+      val csvDir = new File(path, "csv")
+      val cars = data.toDF("make", "comment", "blank")
+      cars.coalesce(1).write.csv(csvDir.getCanonicalPath)
+
+      csvDir.listFiles().filter(_.getName.endsWith("csv")).foreach({ csvFile =>
+        val readBack = Files.readAllBytes(csvFile.toPath)
+        val expected = ("Tesla,,\"\"" + Properties.lineSeparator).getBytes()
+        assert(readBack === expected)
+      })
+    }
```

Comment:
Let's read the written text back by Spark:
```scala
    withTempPath { path =>
      Seq(("Tesla", null: String, "")).toDS().repartition(1).write.csv(path.getCanonicalPath)
      checkAnswer(spark.read.text(path.getCanonicalPath), Row("Tesla,,\"\""))
    }
```

Reply:
Why I read the written with an original file reader is that I want to show other users what changes clearly in saved csv files after this PR.  Before and after this PR, null values are both read as null in DataFrame, maybe someone can not get this change clearly.

But I think your advice is good, so I add another part to read back with Spark.

Reply:
> I want to show other users what changes clearly in saved csv files after this PR.

You will show the same while reading by Spark's **text** datasource. It just splits input by lines. Even the splitting can be avoid using the `wholetext` option.

BTW, in the code `+ Properties.lineSeparator`, you made an assumption that Spark/uniVocity uses the separator in write.

Reply:
Thank you for your explanation, I got it and it's more simple and right.